frontend/src/app/applications-upload/applications-upload.page.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * @description
+ * This page send the applications to the server
+ */
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { NavController } from '@ionic/angular';
+
+import { ToastController } from '../common/ion-wrappers/toast-controller';
+import {
+  EducationInfoEntry,
+  FormsData,
+  KeyCompetenciesEntry,
+} from '../model/forms-data.model';
+import { StorageHandlerService } from '../services/storage-handler.service';
+
+@Component({
+  selector: 'app-applications-upload',
+  templateUrl: './applications-upload.page.html',
+  styleUrls: ['./applications-upload.page.scss'],
+})
+export class ApplicationsUploadPage implements OnInit {
+  constructor(
+    private navController: NavController,
+    private storage: StorageHandlerService,
+    private httpClient: HttpClient,
+    private toastController: ToastController
+  ) {}
+  /**
+   * The total application size.
+   */
+  totalSize = 0;
+  /**
+   * The application size that is being uploaded.
+   */
+  uploadSize = 1;
+  /**
+   * To check whether all the applications are uploaded.
+   */
+  isSuccess = false;
+  /**
+   * To check whether there is error during upload.
+   */
+  isError = false;
+  /**
+   * To display the error message dynamically.
+   */
+  errorMessage: string;
+  /**
+   * URL of the host.
+   */
+  url = 'http://localhost:9090';
+
+  /**
+   * In this method locally stored data are fetched and post request function is called for one application at time.
+   */
+  ngOnInit(): void {
+    this.storage
+      .getAllItems<FormsData>(this.storage.applicantDetailsDb)
+      .then(async (data) => {
```

Comment:
Why it has to be async and await?

Reply:
To wait for the response from the postRequest task to check if there is any error. if there is error it just do not process further applications.